src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -115,4 +122,62 @@ public void removeNullEntryThrowsException() {
         database.removeEntry(null);
     }
 
+    @Test
+    public void insertString() {
+        BibDatabase database = new BibDatabase();
+        assertEquals(Collections.emptySet(), database.getStringKeySet());
+        assertTrue(database.hasNoStrings());
+        BibtexString string = new BibtexString(IdGenerator.next(), "DSP", "Digital Signal Processing");
+        database.addString(string);
+        assertFalse(database.hasNoStrings());
+        assertEquals(database.getStringKeySet().size(), 1);
+        assertEquals(database.getStringCount(), 1);
+        assertTrue(database.getStringValues().contains(string));
+        assertTrue(database.getStringKeySet().contains(string.getId()));
+        assertEquals(string, database.getString(string.getId()));
+    }
+
+    @Test
+    public void insertAndRemoveString() {
+        BibDatabase database = new BibDatabase();
+        BibtexString string = new BibtexString(IdGenerator.next(), "DSP", "Digital Signal Processing");
+        database.addString(string);
+        database.removeString(string.getId());
+        assertTrue(database.hasNoStrings());
+        assertEquals(database.getStringKeySet().size(), 0);
+        assertEquals(database.getStringCount(), 0);
+        assertFalse(database.getStringValues().contains(string));
+        assertFalse(database.getStringKeySet().contains(string.getId()));
+        assertNull(database.getString(string.getId()));
+    }
+
+    @Test
+    public void hasStringLabel() {
+        BibDatabase database = new BibDatabase();
+        BibtexString string = new BibtexString(IdGenerator.next(), "DSP", "Digital Signal Processing");
+        database.addString(string);
+        assertTrue(database.hasStringLabel("DSP"));
+        assertFalse(database.hasStringLabel("VLSI"));
+    }
+
+    @Test(expected = KeyCollisionException.class)
+    public void addSameStringLabelTwice() {
+        BibDatabase database = new BibDatabase();
+        BibtexString string = new BibtexString(IdGenerator.next(), "DSP", "Digital Signal Processing");
+        database.addString(string);
+        string = new BibtexString(IdGenerator.next(), "DSP", "Digital Signal Processor");
+        database.addString(string);
+        fail();
```

Comment:
I will consistently add those as Codacy reports it as an issue. Conceptually you are correct though.